test/sparse/sparse.jl

```diff
@@ -945,886 +1004,951 @@ function test_getindex_algs{Tv,Ti}(A::SparseMatrixCSC{Tv,Ti}, I::AbstractVector,
+end
+
+@testset "Sparse Matrix Broadcasting" begin
+    let  A = sprand(10,10,0.3), B = sprand(10,10,0.3), CF = rand(10,10), AF = Array(A), BF = Array(B), C = sparse(CF)
+        @test A .* B == AF .* BF
+        @test A[1,:] .* B == AF[1,:] .* BF
+        @test A[:,1] .* B == AF[:,1] .* BF
+        @test A .* B[1,:] == AF .*  BF[1,:]
+        @test A .* B[:,1] == AF .*  BF[:,1]
+
+        @test A .* B == AF .* BF
+        @test A[1,:] .* BF == AF[1,:] .* BF
+        @test A[:,1] .* BF == AF[:,1] .* BF
+        @test A .* BF[1,:] == AF .*  BF[1,:]
+        @test A .* BF[:,1] == AF .*  BF[:,1]
+
+        @test A .* B == AF .* BF
+        @test AF[1,:] .* B == AF[1,:] .* BF
+        @test AF[:,1] .* B == AF[:,1] .* BF
+        @test AF .* B[1,:] == AF .*  BF[1,:]
+        @test AF .* B[:,1] == AF .*  BF[:,1]
+
+        @test A .* B == AF .* BF
+        @test A[1,:] .* B == AF[1,:] .* BF
+        @test A[:,1] .* B == AF[:,1] .* BF
+        @test A .* B[1,:] == AF .*  BF[1,:]
+        @test A .* B[:,1] == AF .*  BF[:,1]
+
+        @test A .* 3 == AF .* 3
+        @test 3 .* A == 3 .* AF
+        #@test A[1,:] .* 3 == AF[1,:] .* 3
+        @test all(A[1,:] .* 3 .== AF[1,:] .* 3)
+        #@test A[:,1] .* 3 == AF[:,1] .* 3
+        @test all(A[:,1] .* 3 .== AF[:,1] .* 3)
+        #TODO: simple comparation with == returns false because the left side is a (two-dimensional) SparseMatrixCSC
+        #      while the right side is a Vector
+
+        @test A .- 3 == AF .- 3
+        @test 3 .- A == 3 .- AF
+        @test A .- B == AF .- BF
+        @test A - AF == zeros(AF)
+        @test AF - A == zeros(AF)
+        @test A[1,:] .- B == AF[1,:] .- BF
+        @test A[:,1] .- B == AF[:,1] .- BF
+        @test A .- B[1,:] == AF .-  BF[1,:]
+        @test A .- B[:,1] == AF .-  BF[:,1]
+
+        @test A .+ 3 == AF .+ 3
+        @test 3 .+ A == 3 .+ AF
+        @test A .+ B == AF .+ BF
+        @test A + AF == AF + A
+        @test (A .< B) == (AF .< BF)
+        @test (A .!= B) == (AF .!= BF)
+
+        @test A ./ 3 == AF ./ 3
+        @test A .\ 3 == AF .\ 3
+        @test 3 ./ A == 3 ./ AF
+        @test 3 .\ A == 3 .\ AF
+        @test A .\ C == AF .\ CF
+        @test A ./ C == AF ./ CF
+        @test A ./ CF[:,1] == AF ./ CF[:,1]
+        @test A .\ CF[:,1] == AF .\ CF[:,1]
+        @test BF ./ C == BF ./ CF
+        @test BF .\ C == BF .\ CF
+
+        @test A .^ 3 == AF .^ 3
+        @test 3 .^ A == 3 .^ AF
+        @test A .^ BF[:,1] == AF .^ BF[:,1]
+        @test BF[:,1] .^ A == BF[:,1] .^ AF
+    end
+end
+
+@testset "Empty Matrix Broadcasting" begin
+    @test spzeros(0,0)  + spzeros(0,0)  == zeros(0,0)
+    @test spzeros(0,0)  * spzeros(0,0)  == zeros(0,0)
+    @test spzeros(1,0) .+ spzeros(2,1)  == zeros(2,0)
+    @test spzeros(1,0) .* spzeros(2,1)  == zeros(2,0)
+    @test spzeros(1,2) .+ spzeros(0,1)  == zeros(0,2)
+    @test spzeros(1,2) .* spzeros(0,1)  == zeros(0,2)
+end
+
+@testset "Throws" begin
```

Comment:
Perhaps "error conditions for `reinterpret`, `reshape`, and `squeeze`" rather than simply "Throws"?